scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,349 @@
+        { x = 96,   y = 0,   z = 459 },
+        { x = 103,  y = 0,   z = 459 },
+        { x = 107,  y = 0,   z = 457 },
+        { x = 112,  y = 0,   z = 452 },
+        { x = 114,  y = 0,   z = 446 },
+        { x = 116,  y = 0,   z = 437 },
+        { x = 119,  y = 0,   z = 427 },
+        { x = 120,  y = 0,   z = 418 },
+        { x = 123,  y = 0,   z = 408 },
+        { x = 128,  y = 0,   z = 388 },
+        { x = 131,  y = 0,   z = 383 },
+        { x = 135,  y = 0,   z = 380 },
+        { x = 140,  y = 0,   z = 379 },
+        { x = 147,  y = 0,   z = 379 },
+        { x = 157,  y = 0,   z = 380 },-- first yellow door
+    },
+    [5] =
+    {
+        -- Path from the first yellow door to the basement and back to the first yellow door
+        { x = 185,  y = 0,   z = 380 }, --First Yellow door
+        { x = 193,  y = 0,   z = 390 },
+        { x = 191,  y = 0,   z = 402 },
+        { x = 199,  y = 0,   z = 407 },
+        { x = 207,  y = 0,   z = 408 },
+        { x = 209,  y = 0,   z = 399 },
+        { x = 206,  y = 0,   z = 390 },
+        { x = 209,  y = 0,   z = 384 },
+        { x = 216,  y = 0,   z = 380 },
+        { x = 221,  y = 0,   z = 379 },
+        { x = 239,  y = 0,   z = 379 },
+        { x = 254,  y = 3,   z = 380 },
+        { x = 259,  y = 4,   z = 384 }, -- Top of stairs
+        { x = 259,  y = 11,  z = 415 },
+        { x = 254,  y = 12,  z = 419 },
+        { x = 219,  y = 16,  z = 420 },
+        { x = 219,  y = 16,  z = 387 },
+        { x = 215,  y = 16,  z = 380 },
+        { x = 211,  y = 16,  z = 380 },
+        { x = 142,  y = 16,  z = 380 },
+        { x = 139,  y = 16,  z = 384 },
+        { x = 139,  y = 16,  z = 419 },
+        { x = 145,  y = 16,  z = 427 },
+        { x = 173,  y = 16,  z = 458 },
+        { x = 178,  y = 16,  z = 460 },
+        { x = 210,  y = 16,  z = 460 },
+        { x = 215,  y = 16,  z = 457 },
+        { x = 219,  y = 16,  z = 452 },
+        { x = 219,  y = 16,  z = 442 },
+        { x = 219,  y = 16,  z = 420 },
+        { x = 240,  y = 16,  z = 419 },
+        { x = 258,  y = 12,  z = 417 },
+        { x = 259,  y = 4,   z = 384 },
+        { x = 254,  y = 3,   z = 380 },
+        { x = 239,  y = 0,   z = 379 },
+        { x = 214,  y = 0,   z = 380 },
+        { x = 208,  y = 0,   z = 386 },
+        { x = 208,  y = 0,   z = 398 },
+        { x = 206,  y = 0,   z = 405 },
+        { x = 198,  y = 0,   z = 409 },
+        { x = 191,  y = 0,   z = 405 },
+        { x = 192,  y = 0,   z = 395 },
+        { x = 193,  y = 0,   z = 389 },
+        { x = 187,  y = 0,   z = 382 },
+        { x = 184,  y = 0,   z = 380 },
+        { x = 164,  y = 0,   z = 380 },
+    },
+}
+
+local setZipPath = function(mob, door, label)
+    if door then
+        if door:getAnimation() == xi.anim.OPEN_DOOR then
+            if currentDirection == 0 then -- West to East
+                mob:pathThrough(paths[label + 1], xi.path.flag.COORDS)
+                if label + 1 == 5 then
```

Comment:
In this context, `label` is hard to reason about. This could also be made into an enum

Reply:
I will change `label` to `currPath` to be more descriptive about what it is
I also added an enum `paths` that replaces the checks for 1/2/3/4/5 values
_Note: I changed the table of path nodes from `paths` to `pathNodes`_